src/other/pyodide-loader/PyodideLoader.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
+
+interface PyodideLoaderProps extends React.ComponentProps<any> {
+    url?: string;
+    getPyodide: (pyodide: any) => void;
+    children: React.ReactNode;
+    triggerLoad?: boolean;
+}
+
+declare global {
+    interface Window {
+        loadPyodide: () => Promise<any>;
+        pyodide: any;
+    }
+}
+
+const defaultSourceUrl = "https://cdn.jsdelivr.net/pyodide/v0.24.0/full/pyodide.js";
+
+function PyodideLoader({
+    url = defaultSourceUrl,
+    getPyodide,
+    children,
+    triggerLoad = true,
+}: PyodideLoaderProps) {
+    const [pyodide, setPyodide] = useState<any>(null);
+    const [pyodideInitialized, setPyodideInitialized] = useState(false);
+
+    const createScriptTag = (src: string) => {
+        return new Promise<void>((resolve, reject) => {
+            const script = document.createElement("script");
+            script.src = src;
+            script.onload = () => resolve();
+            script.onerror = () => reject(new Error(`Failed to load script ${src}`));
+            document.body.appendChild(script);
+        });
+    };
+
+    const initializePyodide = async () => {
+        try {
+            await createScriptTag(url);
+            window.pyodide = await window.loadPyodide();
+            const loadedPyodide = window.pyodide;
```

Comment:
Should be assigned to a prop variable

Reply:
Props in React are readonly. The correct approach is to pass a callback function from parent, that will get this pyodide instance after it's fully loaded. We use this pattern everywhere.